Update to v5.15.5-lts and use f-e-d-c

tinywrkb opened the pull request:
* Build the latest 5.15 tagged release from code.qt.io and also bump pciutils to v3.7.0.  
* Add x-checker-data properties to qt5-qtwebengine and pciutils modules.
* Switch qt5-qtwebengine to git source. **update: and to KDE mirror to be able to use the JSON checker.**
* Add SDK extension org.freedesktop.Sdk.Extension.node14.
* ~~I haven't bothered adding the x-checker-data property to python2 as it's not distributed, and I prefer to not touch build dependencies on such a long build.~~ **update:** Also added python2 check now.
* The regex pattern for qt5-qtwebengine seems to match the 5.15.x tags correctly.
* First time updating with flatpak-external-data-checker (manually), later update PRs will ~~hopefully~~ be opened by flathubbot.
* ~~This was not tested as build time is too long.~~ **Update: It's working correctly with qutebrowser.**
* **update: Enabled PipeWire RTC support and the needed PipeWire 0.2 module (pre-Chromium 89 requirement). Wayland screen sharing is now possible with qutebrowser.**

Closes #14.

tinywrkb committed:
use f-e-d-c

Add x-checker-data properties to qt5-qtwebengine and pciutils modules.
Switch qt5-qtwebengine to git source.
Add SDK extension org.freedesktop.Sdk.Extension.node14.

tinywrkb committed:
Update to v5.15.5-lts

Build the latest 5.15 tagged release from code.qt.io.
Also bump pciutils to v3.7.0.
First time updating with flatpak-external-data-checker, later update
PRs will hopefully be opened by flathubbot.
Closes #14.

tinywrkb commented:
Oh! I didn't add Chromium git source. The CI build run will gonna fail early pretty early. I will have to get back to this in another day. edit: Actually, it might work as flatpak-builder should be able to fetch git submodules.

tinywrkb commented:
I had to add nodejs SDK extensions in order to actually have WebEngine enabled in the build. I don't see a major difference in the build config.

_5.15.2 tarball release config_
```
Qt WebEngine Build Tools:
  Use System Ninja ....................... yes
  Use System Gn .......................... no
  Jumbo Build Merge Limit ................ 8
  Developer build ........................ no
  Sanitizer .............................. no
  QtWebEngine required system libraries:
    fontconfig ........................... yes
    dbus ................................. yes
    nss .................................. yes
    khr .................................. yes
    glibc ................................ yes
  QtWebEngine required system libraries for qpa-xcb:
    x11 .................................. yes
    libdrm ............................... yes
    xcomposite ........................... yes
    xcursor .............................. yes
    xi ................................... yes
    xtst ................................. yes
  Optional system libraries used:
    re2 .................................. no
    icu .................................. yes
    libwebp, libwebpmux and libwebpdemux . yes
    opus ................................. yes
    ffmpeg ............................... yes
    libvpx ............................... yes
    snappy ............................... no
    glib ................................. yes
    zlib ................................. yes
    minizip .............................. no
    libevent ............................. no
    jsoncpp .............................. no
    protobuf ............................. no
    libxml2 and libxslt .................. no
    lcms2 ................................ yes
    png .................................. yes
    JPEG ................................. yes
    harfbuzz ............................. yes
    freetype ............................. yes
Qt WebEngineCore:
  Embedded build ......................... no
  Full debug information ................. no
  Pepper Plugins ......................... yes
  Printing and PDF ....................... yes
  Proprietary Codecs ..................... yes
  Spellchecker ........................... yes
  Native Spellchecker .................... no
  WebRTC ................................. yes
  PipeWire over GIO ...................... no
  Geolocation ............................ yes
  WebChannel support ..................... yes
  Kerberos Authentication ................ no
  Extensions ............................. yes
  Node.js ................................ no
  Support qpa-xcb ........................ yes
  Use ALSA ............................... yes
  Use PulseAudio ......................... yes
Qt WebEngineQml:
  Support Qt WebEngine Qml ............... yes
  UI Delegates ........................... yes
  Test Support ........................... no
Qt WebEngineWidgets:
  Support Qt WebEngine Widgets ........... yes
Qt PDF:
  Support V8 ............................. no
  Support XFA ............................ no
  Support XFA-BMP ........................ no
  Support XFA-GIF ........................ no
  Support XFA-PNG ........................ no
  Support XFA-TIFF ....................... no
Qt PDF Widgets:
  Support Qt PDF Widgets ................. yes

WARNING: Building without node.js will disable some features of QtWebEngine DevTools.
```

_5.15.4-lts git tag config_
```
Qt WebEngine Build Tools:
  Use System Ninja ....................... yes
  Jumbo Build Merge Limit ................ 8
  Developer build ........................ no
  Sanitizer .............................. no
  QtWebEngine required system libraries:
    fontconfig ........................... yes
    dbus ................................. yes
    nss .................................. yes
    khr .................................. yes
    glibc ................................ yes
  QtWebEngine required system libraries for qpa-xcb:
    x11 .................................. yes
    libdrm ............................... yes
    xcomposite ........................... yes
    xcursor .............................. yes
    xi ................................... yes
    xproto (glproto) ..................... yes
    xtst ................................. yes
    xkbfile .............................. yes
  Optional system libraries used:
    re2 .................................. no
    icu .................................. yes
    libwebp, libwebpmux and libwebpdemux . yes
    opus ................................. yes
    ffmpeg ............................... yes
    libvpx ............................... yes
    snappy ............................... no
    glib ................................. yes
    zlib ................................. yes
    minizip .............................. no
    libevent ............................. no
    libxml2 and libxslt .................. no
    lcms2 ................................ yes
    png .................................. yes
    JPEG ................................. yes
    harfbuzz ............................. yes
    freetype ............................. yes
Qt WebEngineCore:
  Embedded build ......................... no
  Full debug information ................. no
  Pepper Plugins ......................... yes
  Printing and PDF ....................... yes
  Proprietary Codecs ..................... yes
  Spellchecker ........................... yes
  Native Spellchecker .................... no
  WebRTC ................................. yes
  PipeWire over GIO ...................... no
  Geolocation ............................ yes
  WebChannel support ..................... yes
  Kerberos Authentication ................ no
  Extensions ............................. yes
  Support qpa-xcb ........................ yes
  Use ALSA ............................... yes
  Use PulseAudio ......................... yes
Qt WebEngineQml:
  Support Qt WebEngine Qml ............... yes
  UI Delegates ........................... yes
  Test Support ........................... no
Qt WebEngineWidgets:
  Support Qt WebEngine Widgets ........... yes
Qt PDF:
  Support V8 ............................. no
  Support XFA ............................ no
  Support XFA-BMP ........................ no
  Support XFA-GIF ........................ no
  Support XFA-PNG ........................ no
  Support XFA-TIFF ....................... no
Qt PDF Widgets:
  Support Qt PDF Widgets ................. yes
```

tinywrkb commented:
Test build seems to work correctly with qutebrowser. Only x86_64 was tested.

```
qutebrowser v2.2.3
Git commit: f35f56e-dirty on HEAD (2021-06-01 18:31:44 +0200)
Backend: QtWebEngine 5.15.5, Chromium 87.0.4280.144
Qt: 5.15.3

CPython: 3.8.8
PyQt: 5.15.4

sip: 4.19.25
colorama: 0.4.4
jinja2: 2.11.3
pygments: 2.8.1
yaml: 5.4.1
adblock: 0.4.4
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.4
PyQt5.QtWebKitWidgets: no
pdf.js: 2.8.335 (/app/share/pdf.js/build/pdf.js)
sqlite: 3.32.1
QtNetwork SSL: OpenSSL 1.1.1k  25 Mar 2021

Style: Adwaita::Style
Platform plugin: wayland
OpenGL: None
Platform: Linux-5.12.11-arch1-1-x86_64-with-glibc2.2.5, 64bit
Linux distribution: KDE Flatpak runtime (kde_flatpak)
Frozen: False
Imported from /app/lib/python3.8/site-packages/qutebrowser-2.2.3-py3.8.egg/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/libexec, data path: /usr
```

gasinvein commented on io.qt.qtwebengine.BaseApp.json:
```diff
@@ -69,9 +83,14 @@
+                    "tag": "v5.15.5-lts",
+                    "commit": "9711f64c5082040cb76f6da5ef4a16037dbda08f",
+                    "x-checker-data": {
+                        "type": "git",
```
Please note that the git type checker currently suffers from flathub/flatpak-external-data-checker#154, it'll cause f-e-d-c to open another PR the next day if the current one wasn't merged.

tinywrkb replied:
Well, that would be very wasteful with a such large package, I will change this.

gasinvein replied:
I think you can either keep using tarballs and apply html type checker to it, or use the github git mirror with json checker to query github API.

tinywrkb replied:
I can't use tarballs as there are no tarball releases newer than 5.15.2. Downloading at a tarball archive of a tag from the Github mirror won't help because qtwebkit-chromium will be missing and we need a specific commit and f-e-d-c will not help with this.  
Using the JSON checker with the Github mirror of the official Qt account also won't help because the releases return an empty response, and the response for the list of tags is missing a timestamp.  

It seems like the only way that this can work is to use the JSON checker with KDE's Gitlab mirror. The following seems to work but it should have a regex to match only 5.15.x releases, and I haven't tried yet to build from KDE.

```
"x-checker-data": {
    "type": "json",
    "url": "https://invent.kde.org/api/v4/projects/qt%2Fqt%2Fqtwebengine/repository/tags",
    "tag-query": ".[0].name",
    "version-query": "$tag",
    "timestamp-query": ".[0].commit.created_at"
}
```

gasinvein replied:
I see. Maybe `map(select(.name | test("^v5.15."))) | first` would do?

tinywrkb replied:
I've been thinking about something more like this:

```
"x-checker-data": {
    "type": "json",
    "url": "https://invent.kde.org/api/v4/projects/qt%2Fqt%2Fqtwebengine/repository/tags",
    "tag-query": "first(.[].name | match( \"v5.15[\\\\d.]+-lts|v5.15[\\\\d.]+\" ) | .string)",
    "version-query": "$tag",
    "timestamp-query": ".[] | select(.name==$tag) | .commit.created_at"
}

```

gasinvein replied:
`match()` returns matching pattern. In this case it'll be `v5.15.5`, while the actual tag is `v5.15.5-lts`

tinywrkb replied:
> match() returns matching pattern. In this case it'll be v5.15.5, while the actual tag is v5.15.5-lts

Right, I noticed this and it's why I put `v5.15[\\\\d.]+-lts` before `v5.15[\\\\d.]+`.

tinywrkb committed:
f-e-d-c: don't use git checker and also check python2

The Git checker does not extract timestamps from tags so flathubbot will
create a new pr on every daily run if the previous pr wasn't merged.
See more details in f-e-d-c's bug report https://github.com/flathub/flatpak-external-data-checker/issues/154

To avoid this, switch to the KDE qtwebengine Gitlab mirror that can be
used with the JSON checker.

Also switch to pcituils tarball releases and add a check for python2.

tinywrkb commented:
Building from the KDE mirror works correctly so as far as I concern this can be merged.  

I wanted to fix a typo in the commit message but considering the resources used for every build run and that only admin can cancel a build job, I think I'll wait until the job queue is empty or keep it as it is.

tinywrkb committed:
Enable PipeWire WebRTC

Add PipeWire 0.2.7 and enable the PipeWire WebRTC option to support
screen sharing in Wayland.

tinywrkb commented:
No response yet from the maintainer so I'm using this chance to also enable PipeWire RTC in this PR

aleixpol commented:
Looks great, thanks everyone, sorry for not reviewing earlier.

aleixpol reviewed (approved):
Thanks!

tinywrkb commented:
Thanks for merging!
